### DISCUSSION TITLE
Custom Resource Health/Actions for Flux

### DISCUSSION THREAD

**Discussion by adberger** [argoproj/argo-cd#18888] [Ideas]
Before making unnecessary effort by making a PR with the corresponding functionality:
Would a contribution of [custom resource health checks](https://argo-cd.readthedocs.io/en/stable/operator-manual/health/#custom-health-checks) & [custom resource actions](https://argo-cd.readthedocs.io/en/stable/operator-manual/resource_actions/#custom-resource-actions) for Flux resources to https://github.com/argoproj/argo-cd/tree/master/resource_customizations be accepted or not?

**Comment by crenshaw-dev** [comment]
Sure! If they're good, useful, and intuitive, we'd definitely consider merging.

Input from Flux experts would be appreciated of course to evaluate the actions and health checks.

**Reply by adberger** [reply]
Very nice to hear!
Will try to make a proposing PR soon.

**Reply by adberger** [reply]
part 1: https://github.com/argoproj/argo-cd/pull/18922

**Reply by adberger** [reply]
part 2: https://github.com/argoproj/argo-cd/pull/19082
